requests/adapters.py

```diff
@@ -464,7 +466,9 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                         pool=conn,
                         connection=low_conn,
                         preload_content=False,
-                        decode_content=False
+                        decode_content=False,
+                        enforce_content_length=True,
+                        request_method=request.method
```

Comment:
Unfortunately yes, there's specific logic for a HEAD request in length_remaining and it's not accessible from the response object in urllib3. We're passing this information in manually in urllib3 in `urlopen` but not `from_httplib`.

I'd say this is probably an oversight I made in shazow/urllib3#949 though, I'd be happy to pass the method along there so we're not having to do this manually from outside of urllib3. I'll open a quick amendment patch in a second if that seems acceptable @Lukasa.